src/nifreeze/model/base.py

```diff
@@ -140,7 +140,7 @@ def __init__(self, dataset, predicted=None, **kwargs):
         if self._locked_fit is None:
             raise TypeError("This model requires the predicted map at initialization")
 
-    def fit_predict(self, *_, **kwargs) -> np.ndarray:
+    def fit_predict(self, *_, **kwargs) -> np.ndarray | None:
```

Comment:
This can't be None for this particular model... That said, it'd be better to fix the type annotation on line 107 for consistency.